src/backend/opencl/CMakeLists.txt

```diff
@@ -60,14 +60,39 @@ ENDIF()
 ADD_DEFINITIONS(-DAF_OPENCL
                 -D__CL_ENABLE_EXCEPTIONS)
 
-OPTION(USE_SYSTEM_CLBLAS "Use system clBLAS" OFF)
-IF(USE_SYSTEM_CLBLAS)
-    FIND_PACKAGE(clBLAS REQUIRED)
-ELSE()
-    INCLUDE(build_clBLAS)
+IF(USE_CLBLAST AND USE_CLBLAS)
+    MESSAGE(SEND_ERROR "Cannot use both CLBlast and clBLAS, please select only one of them using USE_CLBLAST=OFF or USE_CLBLAS=OFF")
+ENDIF()
+
+IF(NOT USE_CLBLAST AND NOT USE_CLBLAS)
+    MESSAGE(SEND_ERROR "The OpenCL backend requires either CLBlast or clBLAS, please select one of them using USE_CLBLAST=ON or USE_CLBLAS=ON")
+ENDIF()
+
+IF(USE_CLBLAST)
+    OPTION(USE_SYSTEM_CLBLAST "Use system CLBlast" OFF)
```

Comment:
OK, I moved it (and the clBLAS one) outside of the condition and added a guard to make sure this is only set if USE_CLBLAST is also set.